nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/MultimodeCommand.java

```diff
@@ -148,6 +158,88 @@ protected int executeCommand() throws CommandException, CommandValidationExcepti
         }
     }
 
+    private Prompter basicPrompter() throws IOException {
+        return new BufferedReaderPrompter(new BufferedReader(new InputStreamReader(System.in, Charset.defaultCharset())));
+    }
+
+    @Nonnull
+    private Prompter getPrompter() throws IOException {
+        if (DISABLE_JLINE) {
+            return basicPrompter();
+        }
+        try {
+            Terminal build = TerminalBuilder.builder()
+                    .system(true)
+                    .build();
+            SystemCompleter systemCompleter = new SystemCompleter();
+
+            habitat.getDescriptors(descriptor -> descriptor.getAdvertisedContracts().contains("com.sun.enterprise.admin.cli.CLICommand"))
+                    .forEach(activeDescriptor -> {
+                        String commandName = activeDescriptor.getName();
+                        if (commandName == null) {
+                            return;
+                        }
+                        if (commandName.startsWith("_")) {
```

Comment:
Both here and for remote commands, maybe extract this into a `isInternalCommand` method?